iceoryx_utils/include/iceoryx_utils/posix_wrapper/pthread.hpp

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) 2020 by Apex.AI Inc. All rights reserved.
```

Comment:
I would call this class/header just thread instead of pthread. In this case p = posix and we call the semaphore also semaphore and not psemaphore. It is clear from the namespace that this is a `posix::thread`.